head.go

```diff
@@ -779,6 +779,12 @@ func (h *headChunkReader) Chunk(ref uint64) (chunks.Chunk, error) {
 
 	s.Lock()
 	c := s.chunk(int(cid))
+
+	if c == nil {
```

Comment:
Maybe comment here on when we'd expect this.

Do we need the same check for the series? It could generally be subject to the same race, right?